distributed/remote.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+import subprocess
+import tempfile
+import sys
+
+from tornado import gen
+
+from tornado.ioloop import IOLoop
+
+from distributed.compatibility import unicode
+from distributed.core import Server, write
+from distributed.utils import get_ip
+
+logger = logging.getLogger('distributed.remote')
+
+
+class RemoteClient(Server):
+    def __init__(self, ip=get_ip(), local_dir=tempfile.mkdtemp(prefix='client-'), loop=None, **kwargs):
+        self.ip = ip
+        self.loop = loop or IOLoop.current()
+        self.local_dir = local_dir
+        handlers = {'upload_file': self.upload_file, 'execute': self.execute}
+        super(RemoteClient, self).__init__(handlers, io_loop=self.loop, **kwargs)
+
+    @gen.coroutine
+    def _start(self, port=0):
+        self.listen(port)
+
+    def start(self, port=0):
+        self.loop.add_callback(self._start, port)
+        logger.info("Remote Client is running at {0}".format(port))
```

Comment:
We should probably include both the hostname/ip and the port like `ip:port`.  I often find that for interactive/local use it's very valuable to have a copy-pasteable address print to the screen.